File: benchmarks/src/perl/ack.pl
use Time::HiRes qw(time);
sub ack {
    my ($m, $n) = @_;
    return $n + 1 if $m == 0;
    return ack($m - 1, 1) if $n == 0;
    return ack($m - 1, ack($m, $n - 1));
}
my $start = time();
my $result = ack(3, 6);
my $end = time();
my $ms = ($end - $start) * 1000;
print "Result: $result
";
printf "Time: %.0fms
", $ms;

File: benchmarks/src/python/ack.py
import time

def ack(m, n):
    if m == 0:
        return n + 1
    if n == 0:
        return ack(m - 1, 1)
    return ack(m - 1, ack(m, n - 1))

start = time.time() * 1000
result = ack(3, 6)
end = time.time() * 1000
print(f"Result: {result}")
print(f"Time: {end - start:.0f}ms")

